aten/src/ATen/native/Convolution.cpp

```diff
@@ -265,23 +265,37 @@ at::Tensor convolution(
                           ctx.benchmarkCuDNN(), ctx.deterministicCuDNN(), ctx.userEnabledCuDNN());
 }
 
-
 at::Tensor _convolution(
     const Tensor& input_r, const Tensor& weight_r, const Tensor& bias_r,
     IntList stride_, IntList padding_, IntList dilation_,
     bool transposed_, IntList output_padding_, int64_t groups_,
     bool benchmark, bool deterministic, bool cudnn_enabled) {
 
+  #define ASSIGN_EXPAND_IF_NEEDED(PARAMS, ATTR, LIST_VALUE, EXPECTED_DIM)    \
```

Comment:
Prefer a simple function here. (It can return `std::vector<int64_t>`) Macros are harder to reason about and harder to edit.